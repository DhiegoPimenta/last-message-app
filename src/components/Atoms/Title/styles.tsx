import styled, { css, DefaultTheme } from 'styled-components'
import { TitleProps } from '.'

const wrapperModifiers = {
    normal: (theme: DefaultTheme) => css`
        font-size: 2.4rem;
        font-weight: ${theme.font.normal};
    `,
    large: () => css`
        font-size: 4.8rem;
    `,
    bold: (theme: DefaultTheme) => css`
        font-weight: ${theme.font.bold};
    `,
    extrabold: (theme: DefaultTheme) => css`
        font-weight: ${theme.font.extrabold};
    `,
    fullWidth: () => css`
        width: 100%;
        text-align: center;
    `,
}

export const Title = styled.h1<TitleProps>`
    ${({ theme, size, color, fullWidth, weight }) => css`
        font-weight: bold;
        color: ${theme.colors[color!]};
        ${wrapperModifiers[size!](theme)}
        ${wrapperModifiers[weight!](theme)}
        ${!!fullWidth && wrapperModifiers.fullWidth()}
    `}
`
