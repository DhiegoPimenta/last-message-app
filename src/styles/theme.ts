export default {
    grid: {
        container: '130rem',
        gutter: '3.2rem',
    },
    border: {
        radius: '1rem',
    },
    font: {
        family: `Poppins, Ubuntu, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif`,
        light: 300,
        normal: 400,
        bold: 600,
        extrabold: 800,
        sizes: {
            xsmall: '1.4rem',
            small: '1.8rem',
            medium: '2.4rem',
            large: '3.2rem',
        },
    },
    colors: {
        primary: '#64CBC5',
        secondary: '#0030FF',
        terciary: '#A0DFDC',
        white: '#FFFFFF',
        gray: '#CECECE',
        danger: '#FBA9B0',
        black: '#444444',
        overlay: 'rgba(0, 0, 0, 0.3)',
    },
    spacings: {
        xxsmall: '0.8rem',
        xsmall: '1.6rem',
        small: '2.4rem',
        medium: '3.2rem',
        large: '4.0rem',
        xlarge: '4.8rem',
        xxlarge: '5.6rem',
    },
    layers: {
        base: 10,
        menu: 20,
        overlay: 30,
        modal: 40,
        alwaysOnTop: 50,
    },
    transition: {
        default: '0.3s ease-in-out',
        fast: '0.1s ease-in-out',
    },
} as const
