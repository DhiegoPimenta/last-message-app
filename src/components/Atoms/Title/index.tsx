import { ThemeColors } from 'styles/global'
import * as S from './styles'

export type TitleProps = {
    children: React.ReactNode
    fullWidth?: boolean
    size?: 'normal' | 'large'
    weight?: 'normal' | 'bold' | 'extrabold'
    color?: ThemeColors
}

const Title = ({
    children,
    size = 'normal',
    color = 'primary',
    weight = 'bold',
    fullWidth,
}: TitleProps) => (
    <S.Title size={size} color={color} fullWidth={fullWidth} weight={weight}>
        {children}
    </S.Title>
)

export default Title
