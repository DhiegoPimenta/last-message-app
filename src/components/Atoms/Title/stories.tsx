import { Story, Meta } from '@storybook/react/types-6-0'

import Title, { TitleProps } from '.'

export default {
    title: 'Atoms/Title',
    component: Title,
} as Meta

export const Basic: Story<TitleProps> = (args) => (
    <Title {...args}>Titulo</Title>
)
Basic.args = {
    size: 'normal',
    color: 'primary',
}
