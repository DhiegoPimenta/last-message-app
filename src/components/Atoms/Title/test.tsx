import { screen } from '@testing-library/react'
import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'

import Title from '.'

describe('<Title />', () => {
    it('should render title', () => {
        const { container } = renderWithTheme(<Title>title</Title>)

        expect(screen.getByText(/title/i)).toBeInTheDocument()

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render mediun and primary color as default', () => {
        renderWithTheme(<Title>title</Title>)

        expect(screen.getByText(/title/i)).toBeInTheDocument()
        expect(screen.getByText(/title/i)).toHaveStyle({
            'font-weight': '600',
            'font-size': '2.4rem',
            color: theme.colors.primary,
        })
    })

    it('should render change size', () => {
        renderWithTheme(<Title size="large">title</Title>)

        expect(screen.getByText(/title/i)).toBeInTheDocument()
        expect(screen.getByText(/title/i)).toHaveStyle({
            'font-weight': '600',
            'font-size': '4.8rem',
        })
    })

    it('should render change color', () => {
        renderWithTheme(<Title color="danger">title</Title>)

        expect(screen.getByText(/title/i)).toBeInTheDocument()
        expect(screen.getByText(/title/i)).toHaveStyle({
            'font-weight': '600',
            'font-size': '2.4rem',
            color: theme.colors.danger,
        })
    })

    it('should render full width', () => {
        renderWithTheme(<Title fullWidth>title</Title>)

        expect(screen.getByText(/title/i)).toBeInTheDocument()
        expect(screen.getByText(/title/i)).toHaveStyle({
            width: '100%',
            'text-align': 'center',
        })
    })
})
