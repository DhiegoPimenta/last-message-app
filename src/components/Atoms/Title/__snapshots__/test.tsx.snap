// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render title 1`] = `
.c0 {
  font-weight: bold;
  color: #64CBC5;
  font-size: 2.4rem;
  font-weight: 400;
  font-weight: 600;
}

<h1
  class="c0"
  color="primary"
>
  title
</h1>
`;
